File: common/src/test/java/com/hartwig/actin/treatment/interpretation/FunctionInputMappingTest.java
package com.hartwig.actin.treatment.interpretation;

import static org.junit.Assert.assertTrue;

import com.hartwig.actin.treatment.datamodel.EligibilityRule;
import com.hartwig.actin.treatment.interpretation.composite.CompositeRules;

import org.junit.Test;

public class FunctionInputMappingTest {

    @Test
    public void everyRuleHasInputsConfigured() {
        for (EligibilityRule rule : EligibilityRule.values()) {
            if (!CompositeRules.isComposite(rule)) {
                assertTrue(FunctionInputMapping.RULE_INPUT_MAP.containsKey(rule));
            }
        }
    }
}

File: common/src/test/kotlin/com/hartwig/actin/trial/input/FunctionInputMappingTest.kt
package com.hartwig.actin.trial.input

import com.hartwig.actin.datamodel.trial.EligibilityRule
import com.hartwig.actin.trial.input.composite.CompositeRules.isComposite
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class FunctionInputMappingTest {

    @Test
    fun `Should have configured every rule in terms of having function input`() {
        for (rule in EligibilityRule.values()) {
            if (!isComposite(rule)) {
                assertThat(rule.input).isNotNull
            }
        }
    }
}